src/conn.c

```diff
@@ -3117,6 +3117,9 @@ _processUrlString(natsOptions *opts, const char *urls)
     char        *ptr         = NULL;
     int         len;
 
+    if (urls == NULL)
```

Comment:
I agree that `natsConnection_To()` sounds like the user should provide the URL(s), so if NULL is passed, it is reasonable to fail. But what do you think of simply do processing of `urls` here if not NULL. If NULL, then `if (count == 0) will be true and we will fallback calling natsOptions_SetURL() with NULL, which is OK. The end-result would be that internally the library will use the DEFAULT_URL.

Reply:
I agree that it is common in C to use NULL mainly for optional, but also for default values. I got this issue because in my apps urls comes from a file and the parser returns NULL for options and urls. For options was ok, but not for urls.

I choose to have this fail because in the [documentantion ](https://docs.nats.io/developing-with-nats/connecting/default_server) and also in other places in the [code](https://github.com/nats-io/nats.c/blob/master/src/conn.c#L3088) default URL is always used as  `natsConnection_ConnectTo(&conn, NATS_DEFAULT_URL)`

BTW what is important here is just to guard `strchr(..)` against NULL values.

Let me know what do you think, and I'll change the code accordingly.

Reply:
PR updated to default to DEFAULT_URL when urls=NULL